pkg/cluster/fixssh.go

```diff
@@ -100,14 +101,50 @@ func (m *manager) checkandUpdateNIC(ctx context.Context, resourceGroup string, i
 			}
 		}
 
-		if m.updateNIC(ctx, &nic, nicName, &lb, i, infraID) {
+		if m.updateILBAddressPool(ctx, &nic, nicName, &lb, i, infraID) {
+			m.log.Printf("updating Network Interface %s", nicName)
+			err = m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, nic)
+			if err != nil {
+				return err
+			}
+		}
+
+		var elbName string
+		switch m.doc.OpenShiftCluster.Properties.ArchitectureVersion {
+		case api.ArchitectureVersionV1:
+			elbName = infraID + "-public-lb"
+		case api.ArchitectureVersionV2:
+			elbName = infraID
+		}
+
+		if m.doc.OpenShiftCluster.Properties.ArchitectureVersion == api.ArchitectureVersionV1 {
+			elbv1, err := m.loadBalancers.Get(ctx, resourceGroup, infraID, "")
+			if err != nil {
+				return err
+			}
+			if m.updateV1ELBAddressPool(ctx, &nic, nicName, &elbv1, i, infraID) {
+				m.log.Printf("Updating Network Interface %s", nicName)
+				err = m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, nic)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		elb, err := m.loadBalancers.Get(ctx, resourceGroup, elbName, "")
+		if err != nil {
+			return err
+		}
+
+		if m.updateELBAddressPool(ctx, &nic, nicName, &elb, i, infraID) {
```

Comment:
should this be executed for v1 and v2 architecture?

Reply:
should this be renamed to updateV2ElBAddressPool? or does it indeed need to be executed for both v1 and v2?

Reply:
`updateELBAddressPool` function needs to be executed for both V1 and V2.